### PR TITLE
yabai: 3.3.10 -> 4.0.1

### PR DESCRIPTION
**Pull request by sudosubin**
###### Description of changes

Bump yabai version (`3.3.10` -> `4.0.1`).
Added `yabai-build-symlinks` runCommand as xcode dependency was added to yabai.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/darwin/yabai/default.nix [hunk 1]
```diff
@@ -1,17 +1,24 @@
-{ lib, stdenv, fetchFromGitHub, darwin, xxd }:
+{ lib, stdenv, fetchFromGitHub, runCommand, darwin, xxd }:
 
+let
+  buildSymlinks = runCommand "yabai-build-symlinks" { } ''
+    mkdir -p $out/bin
+    ln -s /usr/bin/xcrun /usr/bin/xcodebuild /usr/bin/tiffutil /usr/bin/qlmanage $out/bin
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/yabai/default.nix:6]: That won't work with the sanbox.

Comment (sudosubin) [reply pkgs/os-specific/darwin/yabai/default.nix:6]: I didn't know about the sandbox concept.

The [`macvim`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vim/macvim.nix) package also uses a very similar configuration to yabai. Would it be ok if I apply the `sandboxProfile` like the one below?

```nix
{
  sandboxProfile = ''
    (allow file-read* file-write* process-exec mach-lookup)
    ; block homebrew dependencies
    (deny file-read* file-write* process-exec mach-lookup (subpath "/usr/local") (with no-log))
  '';
}
```

Comment (SuperSandro2000) [reply pkgs/os-specific/darwin/yabai/default.nix:6]: Hmmmm

We can't build it with the tools in nixpkgs, can we?

Comment (sudosubin) [reply pkgs/os-specific/darwin/yabai/default.nix:6]: When I tried, it couldn't build with other nixpkgs tools like `xcbuild`.

Comment (SuperSandro2000) [reply pkgs/os-specific/darwin/yabai/default.nix:6]: @NixOS/darwin-maintainers

Comment (reckenrode) [reply pkgs/os-specific/darwin/yabai/default.nix:6]: > We can't build it with the tools in nixpkgs, can we?

I tried to make it work, but there are a few problems that stopped me.

* SDK: x86_64-darwin SDK is too old. The minimum supported version of macOS is 10.13.
* clang: requires `ptrauth.h` (arm64e), which is still in the process of being upstreamed (see [D112941][1]) as far as I can tell.

There’s another issue. Yabai requires entitlements. There’s not no good way to do that right now in nixpkgs. The way to go is probably to use upstream’s binary like was down for iTerm.

[1]: https://reviews.llvm.org/D112941

**Pull request closed without merging**